File: repos/koi-backend/flutter_lib/lib/storage/repositories/user_repository.dart
import 'package:hive/hive.dart';
import '../hive_manager.dart';
import '../../models/hive_user_model.dart';

class UserRepository {
  Box<HiveUser> get _box => HiveManager.users;

  Future<HiveUser?> getUser(String id) async => _box.get(id);

  Future<void> saveUser(HiveUser user) async => _box.put(user.id, user);

  Future<void> updateLastActive(String id) async {
    final user = _box.get(id);
    if (user == null) return;
    user.lastActive = DateTime.now();
    await user.save();
  }

  Future<void> deleteUser(String id) async => _box.delete(id);

  List<HiveUser> getAllUsers() => _box.values.toList();

  bool hasUser(String id) => _box.containsKey(id);
}

File: repos/koi-web/src/lib/repositories/user_repository.ts
import { db } from '../db';
import type { User } from '../../models/index';

export class UserRepository {
  async save(user: User): Promise<void> {
    await db.users.put(user);
  }

  async getById(id: string): Promise<User | undefined> {
    return db.users.get(id);
  }

  async updateLastActive(id: string): Promise<void> {
    await db.users.update(id, { lastActive: new Date() });
  }

  async delete(id: string): Promise<void> {
    await db.users.delete(id);
  }

  async getAll(): Promise<User[]> {
    return db.users.toArray();
  }
}

export const userRepository = new UserRepository();
